src/schema-parser/openapi-schema-parser.ts

```diff
@@ -0,0 +1,118 @@
+import { SchemaParser, ParseSchemaInput, ValidateSchemaInput } from "../schema-parser";
+import type { AsyncAPISchema, SchemaValidateResult } from '../types';
+import {Spectral, Document, type RuleDefinition, type RulesetDefinition } from "@stoplight/spectral-core";
+import { Json } from "@stoplight/spectral-parsers";
+import { oas as oasRuleset } from "@stoplight/spectral-rulesets";
+const toJsonSchema = require('@openapi-contrib/openapi-schema-to-json-schema');
+
+export function OpenAPISchemaParser(): SchemaParser {
+  return {
+    validate,
+    parse,
+    getMimeTypes,
+  }
+}
+
+const spectral = new(Spectral)
+let rule = {...oasRuleset};
+// Unnecessary rules
+[
+    'oas3-unused-component',
+    'info-contact',
+    'info-description',
+    'info-license',
+    'oas3-api-servers',
+
+].forEach(function(name) {
+    delete (rule.rules as Record<string, Readonly<RuleDefinition>>)[name];
+});
+spectral.setRuleset(rule as RulesetDefinition);
+
+async function validate(input: ValidateSchemaInput<unknown, unknown>): Promise<SchemaValidateResult[]> {
+    const oas = {"openapi":"3.0.0","info":{"title":"a","description":"a","version":"1"},"paths":{},"components":{"schemas":{"schema":input.data}}};
+    let result: SchemaValidateResult[] = []
+    try {
+        const document = new Document(JSON.stringify(oas), Json);
+        let { results } = await spectral.runWithResolved(document);
+        results.forEach(r => {
+            result.push({
+                message: r.message,
+                path: input.path.concat(r.path.filter(p => !["components", "schemas", "schema"].includes(p.toString()))),
+            });
+        });
+    } catch(err) {
+      console.error(err);
+    } 
+
+    return result;
```

Comment:
I have problem with that 😄 I see that you use Spectral , but I don't think so that it's needed here, because we only should validate schema shape and running spectral will only slow down our validation here. We should use here AJV and validate input data based on that JSON Schema https://github.com/stoplightio/spectral/blob/develop/packages/rulesets/src/oas/schemas/3.0.json#L313

Reply:
We can use spectral and only leave the rule that uses the schema? We could in fact do the same for AsyncAPI Schema?

Reply:
But then you put every rule to the final bundle. We should avoid that. For AsyncAPI case yes, I agree we can reuse spectral but not as separate Spectral instance, but this same.

Reply:
Is it really making a big difference in terms of bundle size? Because in my opinion, it simplifies and unifies the logic by using always the same tool. It also returns the expected errors, etc.

Reply:
That's fair enough 👍 Good point.

In fact, I see like a good feature request for Spectral: to have the possibility to only import few rules from a ruleset instead of all? Do you think this would be possible?

Reply:
It's not possible in this way. You can simply filter rules, but that rules will be still in final bundle.

Reply:
I changed the code and now it's using AJV as you suggested.

Reply:
Thanks!